File: codequest/2019Solutions/Lockheed Solutions/Java/Prob18.java
/* MandelbrotSet.java
 *
 * Copyright (c) 2019 Lockheed Martin (LM) as an unpublished work. 
 * All Rights Reserved.
 *
 * This file does not contain any Lockheed Martin Proprietary Information.
 */

import java.util.Scanner;

public class Prob18 {

	public static void main(String[] args) {
		try (Scanner input = new Scanner(System.in)) {
			int testCases = Integer.parseInt(input.nextLine());

			for (int testcase = 0; testcase < testCases; testcase++) {
				String line = input.nextLine();
				String[] data = line.split(" ");
				
				double a = Double.parseDouble(data[0]);
				double b = Double.parseDouble(data[1]);
				double originalA = a;
				double originalB = b;
				
				String originalC = data[0] + "+" + data[1] + "i ";
				
				double absZ = Math.sqrt((a * a) + (b * b));
				
				int n = 1;
				
				while(n <= 50 && absZ < 100) {
					double newA = (a * a) - (b * b) + originalA;
					double newB = (2 * a * b) + originalB;
					
					a = newA;
					b = newB;
					absZ = Math.sqrt((a * a) + (b * b));
					n++;
				}
				
				if(n <= 10) {
					System.out.println(originalC + "RED");
				}
				else if(n <= 20) {
					System.out.println(originalC + "ORANGE");
				}
				else if(n <= 30) {
					System.out.println(originalC + "YELLOW");
				}
				else if(n <= 40) {
					System.out.println(originalC + "GREEN");
				}
				else if(n <= 50) {
					System.out.println(originalC + "BLUE");
				}
				else {
					System.out.println(originalC + "BLACK");
				}
			}
		}
	}

}

File: codequest/2019Solutions/Lockheed Solutions/C/Prob18.c
#include <string.h>
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <complex.h>

int main()
{
    int testCases;
    scanf("%d\n", &testCases);

    for(int testcase = 0; testcase < testCases; testcase++) {
        char string[100];
        gets(string);

        double real;
        double imag;
        sscanf(string, "%lf %lf\n", &real, &imag);

        complex c = real+I*imag;
        complex z = 0;
        int n = 0;

        while ( (cabsl(z) <= 100) && (n < 51) ) {
            n++;
            z = z*z + c;
        }

        char* space = strstr(string, " ");
        space[0] = '+';

        printf("%si ", string);

        if (n <= 10)
            printf("RED");
        else if (n <= 20)
            printf("ORANGE");
        else if (n <= 30)
            printf("YELLOW");
        else if (n <= 40)
            printf("GREEN");
        else if (n <= 50)
            printf("BLUE");
        else
            printf("BLACK");
        printf("\n");
    }
}
